### DISCUSSION TITLE
neural network degradation with abinit_het

### DISCUSSION THREAD

**Discussion by oleuns** [zhonge/cryodrgn#342] [Show and tell]
Hey!
 
I am currently testing the abinit_het and encountered an issue. I am running on a dataset with 200,000 particles downsampled to 128 with the recommended commands (cryodrgn abinit_het particles.128.mrcs --ctf ctf.pkl --zdim 8 -o) using cryoDRGN 3.0. However, it appears that the network degrades over time and artifacts become visible after iterations 5-6, which become more pronounced with each iteration until the the reconstruction totally degrades. I reproduce this behavior with multiple datasets. The particles are exported from cryoSPARC.

Does anyone know a workaround for this? From the publication, it seems like this is a known issue that can be resolved.
 
Best,
Ole

**Comment by Guillawme** [comment]
Is this specific to version 3? I don't think I have seen what you describe with 2.3. Would you be able to run the same job with 2.3 and determine if this happens too?

**Comment by oleuns** [comment]
I haven't tried it with v2.3 yet, I will give it a shot and compare it to v3.0. Do you use particles exported from cryoSPARC or directly the star file from the WARP-picked particles?

**Comment by Guillawme** [comment]
The places I collect data from don't use Warp. I use particles exported from cryoSPARC. More precisely, I run a new extract job in cryoSPARC with the set of particles I want, with Fourier crop to 128 (or whatever box size I want for cryoDRGN, but I always try 128 first), and then I run `cryodrgn preprocess` on these particles (with the same box size, so no further downsampling happens at this step).

**Comment by oleuns** [comment]
I will try it following your steps using v2.3.0. So far I tested it by exporting particles in cryoSPARC from a homo/NU-refinement and then downsampled them with `cryodrgn downsample` to 128 box size. The rationale behind this was that these particles should be reconstructed into something useful that can be compared to the traditional processing in cryoSPARC.

**Comment by Guillawme** [comment]
I think you can also use `cryodrgn backproject` on the output of `cryodrgn preprocess` to check that the input particle images and metadata are correct.
I don't do it because these days I mostly use `cryodrgn abinit_het` to complement 2D classification, so the initial particles don't have poses necessary for `cryodrgn backproject`.

**Comment by oleuns** [comment]
`cryodrgn backproject` with preprocessed particles using v2.3 and v3.1.0-b beta results in a good-looking reconstruction that closely matches the cryoSPARC reconstruction. This indicates that the particles are fine. One reason might be that the protein I am working on is relatively small (<200 kDa), to check if it's a particle size-related issue I will try it with a larger particle. Another idea that I had was that maybe the "clean" dataset is problematic and causes overfitting.

**Comment by michal-g** [comment]
I've migrated this thread into Discussions as it has shifted to examining the properties of cryoDRGN and input datasets rather than identifying a particular issue, but we are still interested in hearing about your success (or lack thereof) in getting good results using `abinit` on your protein! My idea would be to try different values for the learning rate if you are seeing evidence of overfitting after relatively few iterations.

**Comment by oleuns** [comment]
Thanks for this tip!I will try to adjust the learning rate and see if this helps. Using a dataset from a different protein I was more successful. This protein is bigger than dataset A (only around 120 kDa) and less symmetrical compared to dataset B (around 300 kDa). In my hands, it seems like either very small particles or medium-sized highly symmetrical particles could be prone to overfitting.

**Comment by oleuns** [comment]
Short update, I did not try to adjust the learning rate yet, but using particles that were not classified from the upstream 2D classification helped to prevent overfitting. I used cryoDRGN 2.3.
